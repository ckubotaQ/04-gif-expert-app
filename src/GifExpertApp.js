import React, {useState} from 'react';
import {AddCategory} from './components/AddCategory'
import {GifGrid} from './components/GifGrid'
const GifExpertApp = ({defaultCategories = []}) => {
// const categorys = ['Naruto', 'One Punch', 'Yugioh'];
const [categories, setCategories] = useState(defaultCategories);
// const hamdleAdd = () => {
//     setCategories([...categories, 'Dr.Stone']);
// }
    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
           
            <ol>
               {categories.map((category, i )=>
                   <GifGrid
                   key= {category}
                   category ={category}/>
               )}
            </ol>
        </>

    );
}


export default GifExpertApp
