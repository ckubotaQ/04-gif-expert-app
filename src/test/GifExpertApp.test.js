import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Pruebas en <GifExpertApp />', () => {
    test('Debe mostrar correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    }),
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['digimon', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
        
    })
    
    

    
})
