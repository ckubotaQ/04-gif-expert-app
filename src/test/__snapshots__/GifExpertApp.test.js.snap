// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="digimon"
      key="digimon"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
